Doc/library/sqlite3.rst

```diff
@@ -1434,10 +1434,13 @@ case-insensitively by name:
 Using the connection as a context manager
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Connection objects can be used as context managers
-that automatically commit or rollback transactions.  In the event of an
-exception, the transaction is rolled back; otherwise, the transaction is
-committed:
+With an open transaction, connection objects can be used as context managers
+that automatically commit or rollback transactions.
+If the body of the ``with`` statement finishes without exceptions,
+the transaction is committed.
+If an exception is raised and not caught, the transaction is rolled back.
+
+If there is no open transaction, the context manager is a no-op.
```

Comment:
This sentence suggests that, even if a transaction is opened within the managed context, if there is no pre-existing open transaction at the time of context entry, then the context manager is a no-op. Is this right?

For example, in the [current example in the docs](https://docs.python.org/3/library/sqlite3.html?highlight=sqlite#using-the-connection-as-a-context-manager), would the first context manager be a no-op if the preceding `con.execute(...)` were immediate committed?

```
import sqlite3

con = sqlite3.connect(":memory:")
con.execute("create table lang (id integer primary key, name varchar unique)")
con.commit()  # NEW, EXPLICIT COMMIT PRIOR TO THE CONTEXT MANAGED USE OF con

# Successful, con.commit() is called automatically afterwards
#  !! Does the extra `con.commit()` now make this a no-op? !!
with con:
    con.execute("insert into lang(name) values (?)", ("Python",))

# con.rollback() is called after the with block finishes with an exception, the
# exception is still raised and must be caught
try:
    # !! Is this context manager a no-op unless there's already an open transaction? !!
    with con:
        con.execute("insert into lang(name) values (?)", ("Python",))
except sqlite3.IntegrityError:
    print("couldn't add Python twice")

# Connection object used as context manager only commits or rollbacks transactions,
# so the connection object should be closed manually
con.close()

```

Reply:
Ahh, I read the original wording as "if there is an open transaction *at the point of calling `__enter__`*. Is the behaviour actually if there is an open transaction at the point of calling `__exit__`? 

A

Reply:
> Ahh, I read the original wording as "if there is an open transaction *at the point of calling `__enter__`*. Is the behaviour actually if there is an open transaction at the point of calling `__exit__`? 

Yes. I'll get to it when I'm back at my laptop.

Reply:
We should consider noting explicitly that no transaction is started on enter. Actually, nothing happens on enter.